### DISCUSSION TITLE
Self compiled tasmota and OTA update. Witch file to use ?

### DISCUSSION THREAD

**Discussion by FulvioSpelta** [arendst/Tasmota#20780] [Q&A] [answered]
Hi all, I'm aware of the very noob question but i prefer to be sure about how to proceed.

Till today i always have compiled and flashed (succesfully) tasmota using VSCode (build, erase and upload).
Now i need to use OTA to upgrade a device.
In the pio\build\tasmota32 directory there are 2 fw files:
- firmware.bin
- firmware.factory.bin

My comprehension is this one:
- firmware.bin -> fw tasmota32 only -> this is the file to use for OTA upgrade (i'm going to use "file upload" method)
- firmware.factory.bin -> full fw with partitions and bootloader -> to use for "erase/flash"

Is this correct ?
Thanks

_edit: formatting_

**Comment by sfromis** [comment, marked as answer]
Do not use either of those two files, instead find the correct ones in `build_output`. That should be something like `tasmota32.bin` and `tasmota32.factory.bin`, and you are correct about the distinction between `.factory` and not.

**Reply by FulvioSpelta** [reply]
Found the `<repo root>\build_output\firmware` that contains both files.
Thanks a lot.
